src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaMessageConsumer.cs

```diff
@@ -129,8 +129,24 @@ protected override async Task ActiveAndActualAgentsCompleted(StopContext context
             // It is time to cancel pending tasks as we already drained current pool
             _checkpointTokenSource.Cancel();
 
-            _consumer.Close();
-            _consumer.Dispose();
+            try
+            {
+                _consumer.Close();
+            }
+            catch (Exception e)
+            {
+                LogContext.Error?.Log(e, "Consumer [{MemberId}] close faulted on {Topic}", _consumer.MemberId, _receiveSettings.Topic);
+            }
+
+            try
+            {
+                _consumer.Close();
```

Comment:
Uh... You mean dispose?

Reply:
what do you mean? 

there are 2 calls, Close + Dispose after

Reply:
oh, I see, LOL